src/plugins/socketmon/socketmon.cpp

```diff
@@ -461,6 +463,30 @@ static event_response_t udp_send_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info
 
     sockaddr.sin6_port = __bswap_16(sockaddr.sin6_port);
 
+    addr_t saddr{};
+    if (VMI_FAILURE == vmi_read_addr_va(vmi, udp_info + 0x60, 0, &saddr) || !saddr ||
+        VMI_FAILURE == vmi_read_addr_va(vmi, saddr + 0x10,    0, &saddr) || !saddr ||
+        VMI_FAILURE == vmi_read_addr_va(vmi, saddr,           0, &saddr) || !saddr)
+    {
+        PRINT_DEBUG("[SOCKETMON] UdpSendMessages failed to read saddr\n");
+    }
+    // Get source port offset based on win version.
+    //
+    uint16_t poff{};
+    if (s->build.buildnumber == win_7_sp1_ver)
+        poff = 0x80;
+    else if (s->build.buildnumber == win_10_1803_ver)
+        poff = 0x78;
```

Comment:
Can we collect these offsets in an array at least if they can't be found in a json profile?

Reply:
yes, but I found another bug in udp so I'll add it to this pr later.